packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts

```diff
@@ -53,21 +53,24 @@ export interface StakePoolProviderDependencies extends DbSyncProviderDependencie
 export class DbSyncStakePoolProvider extends DbSyncProvider(RunnableModule) implements StakePoolProvider {
   #builder: StakePoolBuilder;
   #cache: InMemoryCache;
+  #epochLength: number;
   #epochMonitor: EpochMonitor;
   #epochRolloverDisposer: Disposer;
   #paginationPageSizeLimit: number;
   #metadataService: StakePoolExtMetadataService;
 
   constructor(
     { paginationPageSizeLimit }: StakePoolProviderProps,
-    { db, cardanoNode, metadataService, cache, logger, epochMonitor }: StakePoolProviderDependencies
+    { db, cardanoNode, genesisData, metadataService, cache, logger, epochMonitor }: StakePoolProviderDependencies
   ) {
     super({ cardanoNode, db, logger }, 'DbSyncStakePoolProvider', logger);
+
+    this.#builder = new StakePoolBuilder(db, logger);
     this.#cache = cache;
+    this.#epochLength = genesisData.epochLength * 1000;
```

Comment:
Just a note (no action needed, maybe just add an inline comment): epochLength can change, so it should come from `EraSummaries` instead of from `CompactGenesis`, then you would need to look up the length of the specific epoch based on slot number. However it would add a lot of complexity to the queries, so for now this will work just fine.

Reply:
Please add the comment @iccicci